release/releaser.go

```diff
@@ -582,11 +589,19 @@ func (r *releaser) releaseActionRegradeServices(services []flux.ServiceID, msg s
 				}
 
 				namespace, serviceName := service.Components()
-				rc.Instance.LogEvent(namespace, serviceName, "Starting regrade "+cause)
-				specs = append(specs, platform.RegradeSpec{
-					ServiceID:     service,
-					NewDefinition: def,
-				})
+				if serviceName == FluxServiceName {
+					rc.Instance.LogEvent(namespace, serviceName, "Starting regrade (no result expected) "+cause)
+					asyncSpecs = append(asyncSpecs, platform.RegradeSpec{
+						ServiceID:     service,
+						NewDefinition: def,
+					})
+				} else {
+					rc.Instance.LogEvent(namespace, serviceName, "Starting regrade "+cause)
+					specs = append(specs, platform.RegradeSpec{
+						ServiceID:     service,
+						NewDefinition: def,
+					})
+				}
```

Comment:
The decision about whether to treat the release as async could be made higher up and passed to us, that's true. I liked this way around simply because it kept the hack to one file :)